task/06-conditions-n-loops-tasks.js

```diff
@@ -373,10 +537,40 @@ function toNaryString(num, n) {
  *   ['/web/favicon.ico', '/web-scripts/dump', '/webalizer/logs'] => '/'
  */
 function getCommonDirectoryPath(pathes) {
-    throw new Error('Not implemented');
+  let newPathesForCompare = []
+  let newPathesAfterCompare = []
+  let firstSymbolPath = "/"
+
+  pathes.forEach((item, index) => {
+    let sharePathes = []
+
+    if (item[0] !== "/") {
+      firstSymbolPath = ""
+      sharePathes = item.slice(1).split("/")
+    } else sharePathes = item.split("/")
+
+    if (!index) newPathesForCompare = newPathesForCompare.concat(sharePathes)
+    else
+      sharePathes.forEach((item, index2) => {
+        if (item === newPathesForCompare[index2])
+          newPathesAfterCompare[index2] = item
+        else {
+          newPathesForCompare[index2] = item
+          newPathesAfterCompare[index2] = ""
+        }
+      })
+  })
+
+  newPathesAfterCompare = newPathesAfterCompare.filter((item) => item !== "")
+
+  if (newPathesAfterCompare.length)
+    return `${firstSymbolPath}${newPathesAfterCompare.join("/")}/`
+  else if (!newPathesAfterCompare.length && firstSymbolPath === "/")
```

Comment:
You can only use "if" on a new line instead of "else if" because of "return".
Also, the implementation of the method can be simplified.